pkg/common/cns-lib/volume/manager.go

```diff
@@ -1530,3 +1534,134 @@ func (m *defaultManager) QuerySnapshots(ctx context.Context, snapshotQueryFilter
 	}
 	return resp, err
 }
+
+func (m *defaultManager) CreateSnapshot(ctx context.Context, volumeID string, desc string) (string, string, string, *time.Time, error) {
+	log := logger.GetLogger(ctx)
+	err := validateManager(ctx, m)
+	if err != nil {
+		return "", "", "", nil, err
+	}
+	// Set up the VC connection
+	err = m.virtualCenter.ConnectCns(ctx)
+	if err != nil {
+		log.Errorf("ConnectCns failed with err: %+v", err)
+		return "", "", "", nil, err
+	}
+
+	// Construct the CNS SnapshotCreateSpec list
+	var cnsSnapshotCreateSpecList []cnstypes.CnsSnapshotCreateSpec
+	cnsSnapshotCreateSpec := cnstypes.CnsSnapshotCreateSpec{
+		VolumeId: cnstypes.CnsVolumeId{
+			Id: volumeID,
+		},
+		Description: desc,
+	}
+	cnsSnapshotCreateSpecList = append(cnsSnapshotCreateSpecList, cnsSnapshotCreateSpec)
+
+	// Call the CNS CreateSnapshots
+	log.Infof("Calling CnsClient.CreateSnapshots: VolumeID [%q] Description [%q] cnsSnapshotCreateSpecList [%#v]", volumeID, desc, cnsSnapshotCreateSpecList)
+	createSnapshotsTask, err := m.virtualCenter.CnsClient.CreateSnapshots(ctx, cnsSnapshotCreateSpecList)
+	if err != nil {
+		log.Errorf("CNS CreateSnapshots failed from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return "", "", "", nil, err
+	}
+
+	// Get the taskInfo
+	createSnapshotsTaskInfo, err := cns.GetTaskInfo(ctx, createSnapshotsTask)
+	if err != nil || createSnapshotsTaskInfo == nil {
+		log.Errorf("Failed to get taskInfo for CreateSnapshots task from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return "", "", "", nil, err
+	}
+	log.Infof("CreateSnapshots: VolumeID: %q, opId: %q", volumeID, createSnapshotsTaskInfo.ActivationId)
```

Comment:
Would be better if we logged the vc task id as well.

Reply:
Same above.